s3parq/publish_redshift.py

```diff
@@ -27,6 +27,103 @@ def _redshift_name_validator(*args):
         if not response[0]:
             raise ValueError(response[1])
 
+def _get_partitions_for_spectrum(filename: str) -> [str]:
+    '''
+    Turns S3 filepath with partitions into list of only those partitions as strings
+    Args:
+        filename (str): entire filepath for a single file
+    ----
+    Returns:
+        final_partitions (list of strings): these are the partitions for that file
+    --------
+    Example:
+        Args:
+            filename = 'some-path/to/data/zipcode=12345/birth_month=january/final_data_set.parquet
+        ----
+        Returns:
+            final_partitions = ['zipcode=12345', 'birth_month=january']
+    '''
+    filepath = filename.split('/')
+    final_partitions = [_dir for _dir in filepath if '=' in _dir]
+    return final_partitions
+
+
+def _format_partition_strings_for_sql(partitions: [str]) -> [str]:
+    '''
+    Formats a list of S3 partition strings for SQL statements
+    Args:
+        partitions ([str]): list of strings representing the partitions in S3
+    ----
+    Returns:
+        formatted_partitions ([str]): list of the same partitions with quotes
+        to use for SQL
+    --------
+    Example:
+        Args:
+            partitions = ['hamburger=abcd', 'hot_dog=1234']
+        ----
+        Returns:
+            formatted_partitions = ["hamburger='abcd'", "hot_dog='1234'"]
```

Comment:
i know i wrote this, but did we confirm this works with date/datetime data types?  i'm afraid i didn't test those when i tried this

Reply:
Yes, just tested that.